server/py/services/api/api/endpoints/runs.py

```diff
@@ -518,3 +520,45 @@ async def abort_run(
     )
 
     return background_task
+
+
+@router.post(
+    "/projects/{project}/runs/{uid}/push_notifications",
+    status_code=HTTPStatus.OK.value,
+)
+async def push_notifications(
+    project: str,
+    uid: str,
+    auth_info: mlrun.common.schemas.AuthInfo = Depends(deps.authenticate_request),
+    db_session: Session = Depends(deps.get_db_session),
+    iter: int = 0,
+):
+    await (
+        framework.utils.auth.verifier.AuthVerifier().query_project_resource_permissions(
+            mlrun.common.schemas.AuthorizationResourceTypes.run,
+            project,
+            uid,
+            mlrun.common.schemas.AuthorizationAction.update,
+            auth_info,
+        )
+    )
+    run = await run_in_threadpool(
+        services.api.crud.Runs().get_run,
+        db_session,
+        uid,
+        iter,
+        project,
+        mlrun.common.formatters.RunFormat.notifications,
+    )
+    db = db_singleton.get_db()
+    framework.utils.notifications.unmask_notification_params_secret_on_task(
+        db, db_session, run
+    )
+    run_notification_pusher_class = (
+        framework.utils.notifications.notification_pusher.RunNotificationPusher
+    )
+    run_notification_pusher_class(
+        [run],
+        run_notification_pusher_class.resolve_notifications_default_params(),
+    ).push()
```

Comment:
isnt it a coroutine you need to await on?

Reply:
No, it isn't
Exactly like here:
https://github.com/mlrun/mlrun/blob/976d1378b9e2ab6bb8b2c41dac5838a47b3321fe/server/py/services/api/main.py#L780-L783

Reply:
then you run blocking IO on your main event loop